sigar_linux.go

```diff
@@ -317,24 +317,18 @@ func (self *NetIfaceList) Get() error {
 		mtuFile := fmt.Sprintf("%v/class/net/%v/mtu", Sysd, ifaceList[i].Name)
 		macFile := fmt.Sprintf("%v/class/net/%v/address", Sysd, ifaceList[i].Name)
 		linkStatFile := fmt.Sprintf("%v/class/net/%v/carrier", Sysd, ifaceList[i].Name)
-		mtu, err := ioutil.ReadFile(mtuFile)
-		if err == nil {
-			ifaceList[i].MTU, _ = strtoull(string(mtu))
-		}
-		macAddr, err := ioutil.ReadFile(macFile)
-		if err == nil {
-			ifaceList[i].Mac = string(macAddr)
-		}
-		linkStat, err := ioutil.ReadFile(linkStatFile)
-		if err == nil {
-			switch string(linkStat) {
-			case "0":
-				ifaceList[i].LinkStatus = "DOWN"
-			case "1":
-				ifaceList[i].LinkStatus = "UP"
-			default:
-				ifaceList[i].LinkStatus = "UNKNOWN"
-			}
+
+		ifaceList[i].MTU = readUint(readFileLine(mtuFile))
+		ifaceList[i].Mac = readFileLine(macFile)
+
+		linkStat := readFileLine(linkStatFile)
+		switch linkStat {
+		case "0":
+			ifaceList[i].LinkStatus = "DOWN"
+		case "1":
+			ifaceList[i].LinkStatus = "UP"
+		default:
+			ifaceList[i].LinkStatus = "UNKNOWN"
```

Comment:
That's correct - previously `LinkStatus` would be the empty string if there was an error reading the file, and `UNKNOWN` if we hit the switch statement but the value wasn't 0 or 1. Personally I like this better because we end up with only one error state instead of two.

Ideally in order to be more tolerant of partial errors, I think it makes more sense to pass the errors up in the gosigar struct, but its a lot more work for sometimes questionable returns when every integer parsing can result in failure.